trl/trainer/grpo_trainer.py

```diff
@@ -251,7 +239,7 @@ def __init__(
                     f"a `torch.dtype` (e.g., 'float32'), but got {dtype}."
                 )
             # Disable caching if gradient checkpointing is enabled (not supported)
-            config = AutoConfig.from_pretrained(model_id)
+            config = AutoConfig.from_pretrained(model_id, trust_remote_code=self.args.trust_remote_code)
             architecture = getattr(transformers, config.architectures[0])
```

Comment:
When using remote code, the idea is that that model is not included in transformers, right? So maybe you need something like this instead:

```python
if hasattr(transformers, config.architectures[0]):
    architecture = getattr(transformers, config.architectures[0])
    model = architecture.from_pretrained(model_id, **model_init_kwargs)
else:
    model = AutoModelForCausalLM.from_pretrained(model_id, trust_remote_code=True)
```

Reply:
Thanks, you are right. I haven't tested the code path of AutoConfig as I was passing an already loaded model to GRPOTrainer. I will try rewriting it like your code.

Reply:
I fixed it. Now it works when `model` passed to GRPOTrainer is str.